pkg/cloud/services/ec2/subnets.go

```diff
@@ -54,80 +54,66 @@ func (s *Service) reconcileSubnets() error {
 		return err
 	}
 
-	// If the subnets are empty, populate the slice with the default configuration.
-	// Adds a single private and public subnet in the first available zone.
-	if len(existing) < 2 && len(subnets) < 2 {
-		zones, err := s.getAvailableZones()
-		if err != nil {
-			return err
-		}
+	unmanagedVPC := s.scope.VPC().IsUnmanaged(s.scope.Name())
 
-		if len(subnets.FilterPrivate()) == 0 {
-			if s.scope.VPC().IsUnmanaged(s.scope.Name()) {
-				return errors.New("expected at least one private subnet available for use, got 0")
-			}
-
-			subnets = append(subnets, &infrav1.SubnetSpec{
-				CidrBlock:        defaultPrivateSubnetCidr,
-				AvailabilityZone: zones[0],
-				IsPublic:         false,
-			})
+	if len(subnets) == 0 {
+		if unmanagedVPC {
+			// If we have a unmanaged VPC then subnets must be specified
+			errMsg := "no subnets specified, you must specify the subnets when using an umanaged vpc"
+			record.Warnf(s.scope.AWSCluster, "FailedNoSubnets", errMsg)
+			return errors.New(errMsg)
```

Comment:
Isn't this a breaking behavioral change?

Reply:
Yes it is

Reply:
@detiber @randomvariable This should be considered a breaking API change, are we planning to bump to a minor release?

Reply:
I think we might need to for this PR in general.

Reply:
I was thinking about this. I don't think this change is a breaking change to the API, as its only added 2 new optional fields.

But the behaviour has changed in 2 ways for new cluster creation:

- For a managed VPC, default subnet creation has changed to be multi-az
- For a unmanaged VPC, it requires you to explicitly specify the subnet id's you want to use and they must exists. This brings it in line with 'bring-your-own-networking'